File: app/src/main/java/bj/rxjavaexperimentation/model/collectionrelease/Artist.java

package bj.rxjavaexperimentation.model.collectionrelease;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Artist {

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("join")
    @Expose
    private String join;
    @SerializedName("resource_url")
    @Expose
    private String resourceUrl;
    @SerializedName("anv")
    @Expose
    private String anv;
    @SerializedName("tracks")
    @Expose
    private String tracks;
    @SerializedName("role")
    @Expose
    private String role;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getJoin() {
        return join;
    }

    public void setJoin(String join) {
        this.join = join;
    }

    public String getResourceUrl() {
        return resourceUrl;
    }

    public void setResourceUrl(String resourceUrl) {
        this.resourceUrl = resourceUrl;
    }

    public String getAnv() {
        return anv;
    }

    public void setAnv(String anv) {
        this.anv = anv;
    }

    public String getTracks() {
        return tracks;
    }

    public void setTracks(String tracks) {
        this.tracks = tracks;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

}

File: app/src/main/java/bj/vinylbrowser/model/common/Artist.kt
package bj.vinylbrowser.model.common

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created by Josh Laird on 19/05/2017.
 */
data class Artist(var anv: String = "",
                  var id: Int = 0,
                  var join: String = "",
                  var name: String = "",
                  @SerializedName("resource_url") var resourceUrl: String = "",
                  var role: String = "",
                  var tracks: String = "") : Parcelable {
    
    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(anv)
        dest.writeInt(id)
        dest.writeString(join)
        dest.writeString(name)
        dest.writeString(resourceUrl)
        dest.writeString(role)
        dest.writeString(tracks)
    }

    override fun describeContents(): Int {
        return 0
    }
}